ports/esp8266/mpconfigport.h

```diff
@@ -10,12 +14,26 @@
 #define MICROPY_ALLOC_PARSE_RULE_INC    (8)
 #define MICROPY_ALLOC_PARSE_RESULT_INC  (8)
 #define MICROPY_ALLOC_PARSE_CHUNK_INIT  (64)
+
+#ifndef MICROPY_PERSISTENT_CODE_LOAD
 #define MICROPY_PERSISTENT_CODE_LOAD (1)
+#endif
```

Comment:
I think with all these options that have guards added in this file, it'd be much simple to just not configure them at all here.  Instead, the mpconfigboard.h file should enable the features it wants.  So in GENERIC_512K/mpconfigboard.h it'd be pretty much empty.  And in GENERIC/mpconfigboard.h it'd enable all the extra features.

Then this file here would be almost unchanged.

Reply:
That would bring it closer in line with the STM32 port, where features are off by default and enabled in board specific mpconfigboard.h's

Reply:
Updated as per your recommendations.
Defaults in `/py/mpconfig.h`, with common features enabled in `/ports/esp8266/mpconfigboard.h`, and board specific features enabled in `/ports/esp8266/boards/GENERIC/mpconfigboard.h`.

Reply:
Looks good!

Reply:
In the STM32 port, there's an additional `mpconfigboard_common.h`, which sets defaults for boards which do not provide enough config.

Reply:
That extra complexity isn't needed here, at least not now.